### ISSUE TITLE
Halloween not triggering

### ISSUE THREAD

**Issue by Kojirremer** [NyxStudios/TShock#1133]
Since the due date for the start of the halloween event passed and the server seemed the same as usual, I noticed that the event did not trigger at all. Forcing the event to trigger by using /forcehalloween does not work either, even though the "Admin has enabled Halloween mode" did appear.

**Comment by WhiteXZ** [comment]
See https://github.com/NyxStudios/TerrariaAPI-Server/pull/41

**Comment by TomyLobo** [comment]
@Kojirremer 
If the devs allow it (the probably won't), I can give you the TerrariaServer.exe i built from this.
If you have xbuild (I think it comes with mono) and git, you can build it yourself:
````
git clone --recursive --branch v4.3.9 https://github.com/NyxStudios/TShock.git
cd TShock/TerrariaServerAPI
git fetch https://github.com/TomyLobo/TerrariaAPI-Server.git fix-halloween
git cherry-pick FETCH_HEAD
cd ..
xbuild /p:Configuration=Release ./TShockAPI/TShockAPI.csproj
cp ./TShockAPI/bin/Release/TerrariaServer.exe ..
cd ..
````

After that, you'll be back in the directory you started at and you'll have a TerrariaServer.exe in there.
That'll be 4.3.9 plus my patch.

**Comment by Kojirremer** [comment]
It would really help me if you did give me the compiled executable. All of my previous compiling attempts, even following pink_panther's instructions, have ended in failure (and even badly broke my .NET framework installation once haha).
Tried to reach you through your Twitter account, but the link on your profile leads to an unexistant account.
My TShock forum account is Kojiro_S (Twitter account is the same but all lowercase if you don't have a forum account).

**Comment by TomyLobo** [comment]
yeah, .NET seems easily broken...
However, I won't distribute unofficial builds through official channels (such as this one) or to contacts mediated through official channels, unless TShock collaborator says it's ok

**Comment by TomyLobo** [comment]
@Kojirremer They released [4.3.11] (https://tshock.co/xf/index.php?threads/tshock-4-3-11-for-terraria-1-3-0-8-api-1-22.4114/) with the halloween fix.
